fundingmanager.go

```diff
@@ -1500,17 +1517,46 @@ func (f *fundingManager) processFundingLocked(msg *lnwire.FundingLocked,
 // handleFundingLocked finalizes the channel funding process and enables the
 // channel to enter normal operating mode.
 func (f *fundingManager) handleFundingLocked(fmsg *fundingLockedMsg) {
+	defer f.wg.Done()
+
+	// If we are currently in the process of handling a funding locked
```

Comment:
Why is this necessary?

Reply:
This is for the case where the peer is sending two `fundingLocked` messages rapidly after one another. In this case we couldn't check the `RemoteNextRevocation` of the channel, since we had no guarantees we were done handling the first message. This is special for `handleFundingLocked`, since it is run as a goroutine.